### DISCUSSION TITLE
How to make animation rate smooth

### DISCUSSION THREAD

**Discussion by Glacial** [dotnet/maui#12280] [Q&A]
Hi all,
I'm working on a cross-platform app that requires buttery smooth scrolling of large items and I'm encountering a whole bunch of stuttering and hitching with MAUI.
	
There's a little test app here that should help my description a bit:
https://github.com/Glacial/MauiAnimationSmoothnessTest
	
The test app move some items around the screen and measures the time between PropertyChanged events being fired as an animation affects a property. An average is based off the last 100 reading as is shown along with a "low" and "high" time within that 100.
The large bars are animated by moving a fixed number of pixels each update instead of using a delta time. This highlights the variance in frame times more. The progress bar is using a MAUI animation so should be using a delta time solution, but can still appear choppy. The statements below are based on Release mode builds.
	
On Windows, the animation loop is very choppy, unable to maintain a steady 60fps. The stats output by the test app show an average around 16.33ms (good) but wobbles where the gap between animation updates is as low as 0.6ms or as high as 33.4ms. This variance creates notably unstable looking movement. This is without interacting with the app.
	
iOS fares much better, maintaining an average 16.66ms per frame with little variance, that is until you interact. When you interact note that the "High" time can suddenly double and frame skips can be noted. Try dragging around the slider to create a little choppiness. My tests were performed on an iPad Air (5th Gen).
	
Is there anything I can do to improve the animation rates so they look more steady and professional (such as paint to screen myself)? Is there still work for the MAUI team to do to improve this (if so should this be re-posted as an issue)?
	
I've experimented with using a separate PlatformTicker to drive my timings with no improvements (probably because I still can't influence the screen paint time) so I assume the problems are with the underlying platform specific hooks.
	
Thanks

**Comment by akhanalcs** [comment]
Opening it as an issue is a better route.

**Reply by Glacial** [reply]
Thanks, will move it.

**Comment by Glacial** [comment]
Now posted as issue here: https://github.com/dotnet/maui/issues/12385

**Comment by SoumyadipYT-OSS** [comment]
How to make smooth horizontal scrolling in dotnet MAUI?

**Reply by davidortinau** [reply]
Depends on what you're scrolling, what the container view is, what the children are, if they are virtualized, how flat you can make the UI tree, what platform and devices you are targeting... There's no single trick unfortunately. You'll need to do the work and investigation.

**Reply by SoumyadipYT-OSS** [reply]
> Depends on what you're scrolling, what the container view is, what the children are, if they are virtualized, how flat you can make the UI tree, what platform and devices you are targeting... There's no single trick unfortunately. You'll need to do the work and investigation.

Which property is good to implement inside the scroll-view built-in method so it makes it more smooth. If there is another way writing a XAML code to implement more smooth scroll views?
